core/sds-aspect-model-java-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/java/AspectModelJavaUtil.java

```diff
@@ -535,11 +535,15 @@ public static String objectEqualsExpression( final StructureElement element ) {
             .collect( Collectors.joining( " && " ) );
    }
 
-   public static String objectsHashCodeExpression( final StructureElement element ) {
-      return element.getProperties().stream()
+   public static String objectsHashCodeExpression( final StructureElement element, final boolean needsLeadingComma ) {
```

Comment:
Can the logic of adding the leading comma be moved into the template rather than the method?

Reply:
I couldn't find a nice solution as we need to know if there are any elements and only then we need the comma. So we would partially the same logic of finding the elements in the template which seems not much better than.
Or do you see a better way for that?

Reply:
Yes, you would have to have to build another nested `#if`, but I admit that this is not pretty either. So I guess we leave it like that.